### PR TITLE
Rename Agent Mode chats and sessions to tasks

### LINKED ISSUE #639: Rename Agent chats and sessions to tasks in user-facing UI
## Summary

Use **task** terminology consistently throughout Agent mode while keeping **chat** terminology unchanged in Chat mode. Internal implementation names such as “session” do not need to change unless they are user-facing.

## Terminology rules

| Context | Chat mode | Agent mode |
| --- | --- | --- |
| Primary item | Chat | Task |
| Create action | New Chat | New Task |
| Collection/history | Chats / Chat history | Tasks / Task history |
| Untitled placeholder | New chat | New task |
| Destructive action | Delete chat | Delete task |
| Current item references | This chat | This task |

The mode itself remains named **Agent**, including the **Chat | Agent** mode switch.

## Scope

Audit and update all user-facing Agent-mode strings, including:

- Sidebar actions, headings, history lists, and empty states
- New-item flows and untitled placeholders
- Context menus and command-menu entries
- Delete confirmations and other dialogs
- Search, filter, and navigation labels
- Status, error, retry, and completion messages
- Toasts and notifications
- Tooltips, keyboard shortcut descriptions, and accessibility labels
- Settings or help text that refers specifically to Agent-mode items

Use **task** for the durable unit in Agent history. Avoid exposing **session** as a synonym for that unit.

## Compatibility and behavior

- Existing Agent history remains intact and appears as tasks after the terminology change.
- Legacy untitled Agent items display **New task** rather than an old chat/session placeholder.
- Existing generated or user-authored titles are not rewritten.
- Chat mode retains its current chat terminology.
- This is a user-facing language change only; routes, storage, types, and internal session concepts may remain unchanged.
- Wording that intentionally compares both modes may use both terms, such as “Chats and tasks are kept separate.”

## Acceptance criteria

- Creating an item in Agent mode is labeled **New Task**.
- Agent history and Agent item actions consistently use **task/tasks**.
- Delete and confirmation flows identify the affected item as a task.
- Empty, loading, error, retry, and completion states use task-appropriate language.
- Tooltips and assistive-technology labels match the visible terminology.
- Existing Agent items continue to load normally and are presented as tasks.
- Chat-mode UI remains unchanged.
- No user-facing **chat** or **session** references remain in Agent-only UI, except user-authored content or deliberate cross-mode explanations.

## Out of scope

- The Chat | Agent switch component, which is tracked separately.
- Renaming internal session models, APIs, database fields, or routes.
- Combining Chat and Agent histories.
- Changing generated/user-authored task titles.

### PR DESCRIPTION
**Pull request by marksftw**
Closes #639

## Summary

- Rename user-facing Agent Mode chats and sessions to **tasks** across navigation, history, dialogs, settings, MCP controls, accessibility labels, and Maple-owned error strings.
- Preserve Chat Mode wording and Maple’s internal session APIs, storage, runtime flow, and task-creation semantics.
- Add only the small shared-component parameters needed to show **chat** in Chat Mode and **task** in Agent Mode, including routing the sidebar action to each mode’s existing new-item callback.
- Change the default visible Agent title from **New agent session** to **New task**.

## Scope

This PR is a user-facing language change only:

- It does not match, parse, or rewrite user messages, Goose output, framework notices, or ordinary runtime errors.
- It does not queue New Task requests during initialization.
- It does not change folder selection, runtime startup parameters, or task persistence timing.
- The Agent sidebar action uses the existing `createSession()` flow only when Agent settings are unlocked; Chat Mode continues to use its existing `addChat()` flow.

The separate initialization/disabled-state behavior improvement is tracked in #645.

## User impact

Agent Mode consistently presents durable work as tasks while Chat Mode retains chat terminology. Existing Agent history and internal session data remain unchanged.

## Validation

- `bun run typecheck`
- `bun test` — 131 passed
- `bun run lint` — 0 errors, 12 pre-existing warnings
- Prettier checks for the changed frontend files
- `cargo fmt --all -- --check`
- `cargo test` — 123 passed
- `git diff --check`

### CODE REVIEW

File: frontend/src-tauri/src/agent.rs [hunk 1]
```diff
@@ -59,7 +59,13 @@ const MAPLE_GOOSE_PERMISSION_CONFIG: &str = r#"user:
...
 "#;
 const RUN_SHUTDOWN_TIMEOUT: std::time::Duration = std::time::Duration::from_secs(10);
-const DEFAULT_AGENT_SESSION_TITLE: &str = "New agent session";
+const DEFAULT_AGENT_TASK_TITLE: &str = "New task";
+const LEGACY_AGENT_UNTITLED_TITLES: [&str; 4] = [
+    "New agent session",
+    "Agent session",
+    "New chat",
+    "Untitled chat",
+];
```
Comment (AnthonyRonning) [inline frontend/src-tauri/src/agent.rs]: We don't need legacy / backwards compatibility stuff in this new code.

**Review by AnthonyRonning — CHANGES_REQUESTED** [review]

File: frontend/src-tauri/src/agent.rs (hunk 1 above)
Comment (marksftw) [reply frontend/src-tauri/src/agent.rs]: Should be resolved now.

Comment (marksftw) [reply frontend/src-tauri/src/agent.rs]: oh shoot, resolving branch conflicts real quick

**Comment by AnthonyRonning** [comment]
Needs rebase.

**Comment by marksftw** [comment]
> Needs rebase.

yeah I caught that just as I hit the review button

**Review by AnthonyRonning — CHANGES_REQUESTED** [review]
Requesting changes because this PR has expanded substantially beyond the terminology change described in #639.

The intended scope is to change user-facing Agent terminology from “chat/session” to “task,” while leaving internal session APIs, storage, and existing behavior unchanged.

This PR currently includes two additional categories of work:

1. It changes New Task behavior, including initialization queuing, folder-selection flow, runtime startup parameters, and when an empty task is persisted.
2. It introduces string matching and rewriting for messages and errors emitted by Goose.

The New Task behavior may be fixing a legitimate problem, but it should not be bundled into a terminology PR. Please revert those behavioral changes here. If the current New Task flow is broken or inadequate, document the existing behavior, the desired behavior, and the relevant edge cases in a separate issue, then address it in a focused PR with direct tests.

I also do not want Maple matching exact Goose prose and rewriting it. This is brittle, couples Maple to Goose’s English wording, and risks modifying content based on text rather than a structured signal. The tests confirm the string-matching implementation, but they do not make that integration boundary maintainable.

Please remove the Goose string-rewriting logic from this PR. If Goose exposes user-facing session terminology that Maple needs to translate, handle that separately through a structured design—for example, typed error/notice categories from Goose, or a single adapter that maps Goose events into Maple-owned enums and presentation copy. That design deserves its own issue and PR.

For this PR, please keep the changes narrowly limited to:

* Static user-facing Agent labels, descriptions, dialogs, tooltips, and accessibility text.
* Small shared-component parameters needed to display “chat” in Chat Mode and “task” in Agent Mode.
* Maple-owned user-facing error strings where changing the literal at its source is straightforward.
* The default visible Agent title changing from “New agent session” to “New task.”

Please do not include runtime-flow changes, new task-creation semantics, initialization state handling, or rewriting of unstructured Goose content. Once the PR is reduced to that scope, I’ll review it again.

**Comment by marksftw** [comment]
Removed the two feature-specific runtime text rewriters so this PR remains a static Maple UI terminology change and does not scan or alter user or Goose text.

What changed:
- Removed the Rust framework-notice transformer that searched assistant and system messages for exact Goose phrases and changed session to task.
- Removed the frontend runtime-error transformer that searched ordinary errors for session phrases.
- User messages, Goose assistant and system text, and ordinary non-MCP errors now pass through without terminology rewriting. The existing system-notice display length limit is unchanged.
- Kept the pre-existing MCP connection-error sanitizer, which classifies and cleans up MCP connection failures rather than performing Task/Session terminology rewriting.
- External messages may therefore continue to say session, including messages such as Session not found.

Validation: cargo test (123 passed), frontend tests, type checking, formatting, and linting all passed. This is included as the separate commit b4719d4.

**Comment by AnthonyRonning** [comment]
Thanks—removing both runtime text rewriters addresses my Goose/error-mapping concern.

The other requested scope change is still outstanding. The PR still contains the New Task behavioral changes, including:

- `isNewTaskRequested` and the initialization-time queue
- the effect that processes a deferred New Task request
- returning a path from `chooseProjectRoot`
- passing an explicit project root through `startRuntime` and `createSession`
- creating a persisted task immediately after folder selection
- keeping the New Task action enabled while initialization is running

Those are functional changes, not terminology changes. Please remove them from this PR and restore the previous initialization, folder-selection, runtime-start, and session-creation behavior.

I understand that the shared sidebar needs a mode-aware label and may need a small callback parameter so a button labeled “New Task” does not execute Chat Mode’s `addChat()` behavior. That narrow shared-component adjustment is fine. The new deferred-creation state machine and folder-selection semantics are not.

If the existing New Task action has a behavioral bug, please document it separately with the current behavior, desired behavior, and edge cases, then address it in a focused PR with direct tests.

Please also update the PR description after removing this behavior; it currently still advertises initialization queuing and the disabled-state fix. Once that remaining behavior is separated, the terminology PR will match the requested scope.

**Comment by marksftw** [comment]
Updated: the latest head removes the remaining New Task behavior expansion so this PR is limited to the requested terminology change.

What is now in place:
- Removed `isNewTaskRequested`, the deferred-request effect, and all initialization-time queuing.
- Restored `chooseProjectRoot`, `startRuntime`, and `createSession` to their previous signatures and flow; there is no explicit-root threading or task creation after folder selection.
- Restored initialization locking, so **New Task** is disabled while Agent settings are initializing or otherwise locked.
- Kept only the narrow shared Sidebar callback allowed in the review: Chat Mode uses its existing `addChat()` path, while Agent Mode uses its existing `createSession()` path once unlocked.
- Removed both Goose/runtime text rewriters; user messages, Goose output, and ordinary runtime errors are not terminology-scanned or rewritten.
- Updated the PR description so it no longer advertises queuing or the disabled-state fix.

The separate transition/disabled-state improvement, including the current behavior, desired behavior, edge cases, persistence decision, and required direct tests, is tracked in #645.

Relevant commits: `570b759` and `4a84ffc`.

Validation: 131 frontend tests and 123 Rust tests passed, along with type checking, formatting, `git diff --check`, and lint with 0 errors (12 pre-existing warnings).

**Review by AnthonyRonning — APPROVED** [review]
The requested scope changes are now addressed. The Goose/runtime string rewriting is removed, the New Task behavior work is separated into #645, and this PR is appropriately limited to user-facing terminology plus the minimal mode-aware sidebar routing.

Approved. Please squash and merge this PR.
